### PR TITLE
Feature dynamische queries

### LINKED ISSUE #43: feature - dynamische OSM-Queries
Für übersichtlichen Code soll die Möglichkeit dynamischer Queries erstellt werden. Diese Möglichkeit kann mittels einer oder mehreren Klassen implementiert werden. Interessant wäre zudem dieses Plugin:

- https://pub.dev/packages/osm_nominatim

welches unsere Funktionalität möglicherweise schon liefert.

### LINKED ISSUE #48: bug - Positionsaktualisierung
Aktuell ist die automatische Positionsermittlung im Hintergrund nicht möglich, das sollte behoben werden.

### PR DESCRIPTION
**Pull request by maximilianley**
dynamische Funktionalität für OSM-Queries implementiert. Parameter:
- Suchbereich (search area)
- eigene momentane Position
- Suchradius
können zu Laufzeit gesetzt werden.
(In Zukunft kann die OSM-Anfrage auch bei Aufruf der Map, oder sogar regemäßig im Hintergrund, stattfinden, um die Position und entsprechende Suchdaten automatisch zu aktualisieren).

Issue für die "dynamische Positionsaktualisierung" ist damit auch umgesetzt.

### CODE REVIEW

File: taesch/lib/api/map_api_logic/querying_tools.dart [hunk 1]
```diff
@@ -9,10 +12,56 @@ class OSMQueries {
...
+    query += "data=[out:json][timeout:50];area[name=\"";// <- may alter timeout variable
+    query+=locationName;
+    query+="\"]->.searchArea;(nwr[\"shop\"=\"supermarket\"](around:";
+    query+="$searchRadius,";
+    query+="$latitude,";
+    query+=longitude;
+    query+=")(area.searchArea););out;";
+
+    //return query1Heilbronn;
+    return query;
```
Comment (sorrowcode) [inline taesch/lib/api/map_api_logic/querying_tools.dart]: Hier wäre es besser einen String zu schreiben und die Werte als Interpolation hinzuzufügen

File: taesch/test/main_test.dart [hunk 2]
```diff
@@ -333,6 +336,36 @@ void main() {
...
+      repository.setPosition(samplePosition);
+      expect(repository.queries.getCustomLocation().latitude, myLat);
+    });
+    test('set search area', (){
+      QueryLocation myLocation = QueryLocation.neckarsulm;
+      repository.queries.setQueryLocation(myLocation);
+      expect(repository.queries.getQueryLocation(), myLocation);
+    });
+    // also attempt query
+  });
```
Comment (sorrowcode) [inline taesch/test/main_test.dart:394]: hier gibt es vielleicht Edge cases, wäre gut, wenn diese ebenfalls getestet würden

**Review by sorrowcode — DISMISSED** [review]

File: taesch/lib/api/map_api_logic/querying_tools.dart (hunk 1 above)
Comment (sorrowcode) [reply taesch/lib/api/map_api_logic/querying_tools.dart]: Hier ein Beispiel für Interpolation
https://dart.dev/codelabs/dart-cheatsheet#string-interpolation

Comment (jonycoo) [reply taesch/lib/api/map_api_logic/querying_tools.dart]: resolved

**Review by sorrowcode — APPROVED** [review]
